### ISSUE TITLE
ghcjs build is failing with out-of-memory

### ISSUE THREAD

**Issue by harendra-kumar** [composewell/streamly#1208]
The issue seems to be with `Streamly.Internal.Unicode.Stream` module with the latest commit (c9232b0e25d895d268eb9f38a6b1fa0e01b75dc6) in that file. On the previous commit it works.

**Comment by adithyaov** [comment]
Related to https://github.com/composewell/streamly/issues/1193

**Comment by harendra-kumar** [comment]
We need to test at commits between 9a253dffd8b4a320dd200ef8e3fdfcd18483aee7 https://github.com/composewell/streamly/commit/c9232b0e25d895d268eb9f38a6b1fa0e01b75dc6 . We can bisect to find out the real culprit.

**Comment by harendra-kumar** [comment]
Looks like some template haskell linking is going on when it hangs:

```
[113 of 120] Compiling Streamly.Internal.Unicode.Stream ( src/Streamly/Internal/Unicode/Stream.hs, /home/harendra/composewell/streamly1/dist-newstyle/build/x86_64-linux/ghcjs-8.4.0.1/streamly-0.8.0/build/Streamly/Internal/Unicode/Stream.js_o ) [Streamly.Data.Array.Foreign changed]
Linking Template Haskell (Streamly.Data.Array,Streamly.Data.Array.Foreign,Streamly.Data.Fold,Streamly.Data.Fold.Tee,Streamly.Data.Prim.Array,Streamly.Data.SmallArray,Streamly.Data.Unfold,Streamly.FileSystem.Handle,Streamly.Internal.BaseCompat,Streamly.Internal.Control.Concurrent,Streamly.Internal.Control.Exception,Streamly.Internal.Control.Monad,Streamly.Internal.Data.Array,Streamly.Internal.Data.Array.Foreign,Streamly.Internal.Data.Array.Foreign.Mut,Streamly.Internal.Data.Array.Foreign.Mut.Type,Streamly.Internal.Data.Array.Foreign.Type,Streamly.Internal.Data.Array.Prim,Streamly.Internal.Data.Array.Prim.Mut.Type,Streamly.Internal.Data.Array.Prim.Pinned,Streamly.Internal.Data.Array.Prim.Pinned.Mut.Type,Streamly.Internal.Data.Array.Prim.Pinned.Type,Streamly.Internal.Data.Array.Prim.Type,Streamly.Internal.Data.Array.Stream.Fold.Foreign,Streamly.Internal.Data.Array.Stream.Foreign,Streamly.Internal.Data.Array.Stream.Mut.Foreign,Streamly.Internal.Data.Atomics,Streamly.Internal.Data.Binary.Decode,Streamly.Internal.Data.Cont,Streamly.Internal.Data.Either.Strict,Streamly.Internal.Data.Fold,Streamly.Internal.Data.Fold.SVar,Streamly.Internal.Data.Fold.Tee,Streamly.Internal.Data.Fold.Type,Streamly.Internal.Data.IOFinalizer,Streamly.Internal.Data.IORef.Prim,Streamly.Internal.Data.List,Streamly.Internal.Data.Maybe.Strict,Streamly.Internal.Data.Parser,Streamly.Internal.Data.Parser.ParserD,Streamly.Internal.Data.Parser.ParserD.Tee,Streamly.Internal.Data.Parser.ParserD.Type,Streamly.Internal.Data.Parser.ParserK.Type,Streamly.Internal.Data.Pipe,Streamly.Internal.Data.Pipe.Type,Streamly.Internal.Data.Producer,Streamly.Internal.Data.Producer.Source,Streamly.Internal.Data.Producer.Type,Streamly.Internal.Data.SVar,Streamly.Internal.Data.SVar.Dispatch,Streamly.Internal.Data.SVar.Pull,Streamly.Internal.Data.SVar.Type,Streamly.Internal.Data.SVar.Worker,Streamly.Internal.Data.Sink,Streamly.Internal.Data.Sink.Type,Streamly.Internal.Data.SmallArray,Streamly.Internal.Data.SmallArray.Type,Streamly.Internal.Data.Stream.Ahead,Streamly.Internal.Data.Stream.Async,Streamly.Internal.Data.Stream.IsStream,Streamly.Internal.Data.Stream.IsStream.Combinators,Streamly.Internal.Data.Stream.IsStream.Common,Streamly.Internal.Data.Stream.IsStream.Eliminate,Streamly.Internal.Data.Stream.IsStream.Enumeration,Streamly.Internal.Data.Stream.IsStream.Exception,Streamly.Internal.Data.Stream.IsStream.Expand,Streamly.Internal.Data.Stream.IsStream.Generate,Streamly.Internal.Data.Stream.IsStream.Lift,Streamly.Internal.Data.Stream.IsStream.Reduce,Streamly.Internal.Data.Stream.IsStream.Top,Streamly.Internal.Data.Stream.IsStream.Transform,Streamly.Internal.Data.Stream.IsStream.Types,Streamly.Internal.Data.Stream.Parallel,Streamly.Internal.Data.Stream.Prelude,Streamly.Internal.Data.Stream.SVar.Eliminate,Streamly.Internal.Data.Stream.SVar.Generate,Streamly.Internal.Data.Stream.Serial,Streamly.Internal.Data.Stream.StreamD,Streamly.Internal.Data.Stream.StreamD.Eliminate,Streamly.Internal.Data.Stream.StreamD.Exception,Streamly.Internal.Data.Stream.StreamD.Generate,Streamly.Internal.Data.Stream.StreamD.Lift,Streamly.Internal.Data.Stream.StreamD.Nesting,Streamly.Internal.Data.Stream.StreamD.Step,Streamly.Internal.Data.Stream.StreamD.Transform,Streamly.Internal.Data.Stream.StreamD.Type,Streamly.Internal.Data.Stream.StreamDK,Streamly.Internal.Data.Stream.StreamDK.Type,Streamly.Internal.Data.Stream.StreamK,Streamly.Internal.Data.Stream.StreamK.Type,Streamly.Internal.Data.Stream.Zip,Streamly.Internal.Data.Time,Streamly.Internal.Data.Time.Clock,Streamly.Internal.Data.Time.Clock.Type,Streamly.Internal.Data.Time.TimeSpec,Streamly.Internal.Data.Time.Units,Streamly.Internal.Data.Tuple.Strict,Streamly.Internal.Data.Unfold,Streamly.Internal.Data.Unfold.Enumeration,Streamly.Internal.Data.Unfold.SVar,Streamly.Internal.Data.Unfold.Type,Streamly.Internal.FileSystem.Dir,Streamly.Internal.FileSystem.FDIO,Streamly.Internal.FileSystem.File,Streamly.Internal.FileSystem.Handle,Streamly.Internal.Foreign.Malloc,Streamly.Internal.Network.Socket,Streamly.Internal.Ring.Foreign,Streamly.Internal.System.IO,Streamly.Internal.System.IOVec,Streamly.Internal.System.IOVec.Type,Streamly.Internal.Unicode.Char,ThRunner1)
```

Found some links that may be relevant, this seems like a known issue with ghcjs:

* https://github.com/ghcjs/ghcjs/issues/449
* https://github.com/ghcjs/ghcjs/issues/668
* https://mail.haskell.org/pipermail/ghc-devs/2014-July/005415.html

**Comment by harendra-kumar** [comment]
Fixed by #1222
